src/test/java/com/puppycrawl/tools/checkstyle/internal/TestUtils.java

```diff
@@ -56,4 +66,46 @@ public static PackageObjectFactory getPackageObjectFactory() throws CheckstyleEx
         final Set<String> packageNames = PackageNamesLoader.getPackageNames(cl);
         return new PackageObjectFactory(packageNames, cl);
     }
+
+    /**
+     * Finds node of specified type among root children on any deep level.
+     * @param root      DetailAST
+     * @param predicate predicate
+     * @return {@link Optional} of {@link DetailAST} node which matches the predicate.
+     */
+    public static Optional<DetailAST> findTokenInASTByPredicate(DetailAST root,
+                                                                Predicate<DetailAST> predicate) {
+        DetailAST curNode = root;
+        while (!predicate.test(curNode)) {
```

Comment:
What prevents this loop from going beyond the root node into it's siblings if `predicate` never returns `true`?